### DISCUSSION TITLE
Access Pinia Store globally

### DISCUSSION THREAD

**Discussion by sdoerger** [vuejs/pinia#739] [Help and Questions] [answered]
I just tried out pinia and so far I like it and would replace with it vuex in an app.

In Vue I access sometimes getters or actions dynamically, via a string. Which I like.

i. e.: A class, which checks if a getter ist true and returns it value
```
import Module from '@/store';

export class computedGetter {
...
const storeGetter = Module.getters[String(fromVar)];
...
}
```

So in a component, in setup, I can do:
`const singleDate = new computedGetter('getCounter');`

OR dispatch an action:
```
import Module from '@/store';

export class DateWatcher {
...
Module.dispatch(stringFromVar, fromAVar);
...
}
```

I tried that with pinia, but stuck by not getting values:
```
import pinia from '@/store';
// or import { pinia } from '@/store';
// Actual file is in '@/store/modules/counter.ts

  export default defineComponent({
    setup() {
      const test = computed(() => {
        if (pinia) {
          console.log('IN TEST');
          
          console.log(pinia);
          // console.log(pinia.state.value['counterId']);
          
          return pinia.state.value[couterId];
        }
      });
      return  {
           test
      }
}
```

Is there a way, to do that?
I hope my explanation is not to sloppy

**Comment by posva** [comment, marked as answer]
You can call `useOtherStore()` inside of your store:

```ts
defineStore('a', () => {
  const other = useOtherStore()
})
```

or https://pinia.esm.dev/core-concepts/getters.html#accessing-other-stores-getters for options api

**Comment by sdoerger** [comment]
Thanks for quick reply. Acutally I thought more of an auto import for all store files. But I just saw, that I used in vuex as well a function to do that. I do it now via   `const modules = import.meta.glob('/src/store/modules/*.ts');` and currenty loop thorugh it, until I get the desired action from a var.
I dunno where, but I thought I saw somewhere, that pinia does that out of the box and I can hustle free access the whole store with its modules through a single import.

But this works for me, so I guess I switch now to pinia :)

**Reply by posva** [reply]
You need to import each individual store that is used so the dependency graph can be automatically created and your code is correctly split. That was possible in vuex with dynamic modules but was not ergonomic

**Reply by sdoerger** [reply]
Thanks for your explanation.
Would you say, my solution via `const modules = import.meta.glob('/src/store/modules/*.ts')` could cause trouble and I better register them per hand?

**Reply by posva** [reply]
Yes, but as always it depends 😅
